infrastructure/main.tf

```diff
@@ -36,37 +36,83 @@ locals {
-      subnet_private_access = var.subnetwork_private_access
-      description           = var.subnetwork_description
-    }
-  ]
-
+  subnets = [for subnet_region, subnet_ip in zipmap(var.subnetwork_regions, local.subnetwork_cidr_ips) : {
+    subnet_name           = var.subnetwork_name
+    subnet_ip             = subnet_ip
+    subnet_region         = subnet_region
+    subnet_private_access = var.subnetwork_private_access
+    description           = var.subnetwork_description
+  }]
   #secondary_ranges = var.network_secondary_ranges
   #firewall_rules = var.firewall_rules
 }
 
+// TODO: Migrate to terraform-google-modules/sql-db/google//modules/private_service_access (below)
+// once https://github.com/terraform-google-modules/terraform-google-sql-db/issues/585 is resolved.
+// We define a VPC peering subnet that will be peered with the
+// Cloud SQL instance network. The Cloud SQL instance will
+// have a private IP within the provided range.
+// https://cloud.google.com/vpc/docs/configure-private-services-access
+
+resource "google_compute_global_address" "google-managed-services-range" {
+  count         = var.create_network ? 1 : 0
+  project       = var.project_id
+  name          = "google-managed-services-${var.network_name}"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = local.network_self_link
+}
+
+# Creates the peering with the producer network.
+resource "google_service_networking_connection" "private_service_access" {
+  count                   = var.create_network ? 1 : 0
+  network                 = local.network_self_link
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.google-managed-services-range[0].name]
+  # This will enable a successful terraform destroy when destroying CloudSQL instances
+  deletion_policy = "ABANDON"
+}
+
+// TODO: Migrate to using the below module block instead of
+// the above "google_compute_global_address" and "google_service_networking_connection" resources
+// once https://github.com/terraform-google-modules/terraform-google-sql-db/issues/585 is resolved.
+// module "private-service-access" {
+//  source  = "terraform-google-modules/sql-db/google//modules/private_service_access"
+//  version = "~> 18.0"
+//  count   = var.create_network ? 1 : 0
+//
+//  project_id      = var.project_id
+//  vpc_network     = var.network_name
+//  # This will enable a successful terraform destroy when destroying CloudSQL instances
+//  deletion_policy = "ABANDON"
+// }
+
 locals {
-  network_name    = var.create_network ? module.custom-network[0].network_name : var.network_name
-  subnetwork_name = var.create_network ? module.custom-network[0].subnets_names[0] : var.subnetwork_name
-  region          = length(split("-", var.cluster_location)) == 2 ? var.cluster_location : ""
-  regional        = local.region != "" ? true : false
+  network_name      = var.network_name
```

Comment:
why we removed this to normal vartype?

Reply:
Is there an advantage to using the output vs. the input variable? My understanding is that this resource just reflects the input to the output (through google_compute_network name argument: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network), so I don't see a good reason to use a more complicated expression here.

Reply:
okay its always better to use module outputs.. as this helps terraform to prepare dependency graphs and prepare better apply sequence..

so rather than adding explicts `depends_on` on network module  is better to use module outputs that terraform considers as implicy dependency..

Reply:
Got it, thanks for the explanation. Changed this back, and removed the `depends_on`